eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/ChannelRelay.scala

```diff
@@ -261,7 +267,7 @@ class ChannelRelay private(nodeParams: NodeParams,
         RelayFailure(CMD_FAIL_HTLC(add.id, Right(AmountBelowMinimum(payload.amountToForward, channelUpdate)), commit = true))
       case Some(channelUpdate) if r.expiryDelta < channelUpdate.cltvExpiryDelta =>
         RelayFailure(CMD_FAIL_HTLC(add.id, Right(IncorrectCltvExpiry(payload.outgoingCltv, channelUpdate)), commit = true))
-      case Some(channelUpdate) if r.relayFeeMsat < nodeFee(channelUpdate, payload.amountToForward) =>
+      case Some(channelUpdate) if r.relayFeeMsat < nodeFee(channelUpdate, payload.amountToForward) && !usePrevFees =>
```

Comment:
I don't think you should define `usePrevFees` far from where it's used, it fits in one line here.

```suggestion
      case Some(channelUpdate) if r.relayFeeMsat < nodeFee(channelUpdate, payload.amountToForward) &&
        outgoingChannel_opt.forall(c => TimestampSecond.now() - c.channelUpdate.timestamp > nodeParams.enforceDelay || r.relayFeeMsat < nodeFee(c.prevChannelUpdate, payload.amountToForward)) =>
```

Reply:
Oh nice. I like how the logic mirrors the current channel update also.

Reply:
I had to add a `flatmap` because `nodeFee` does not take an `Option`. It's a bit less elegant but I couldn't find a better way to get it working.  Fixed in [a302f32](https://github.com/ACINQ/eclair/pull/2201/commits/a302f32f5cd3a7c22c22b342e8a54fe09ad15872).

Reply:
You're right, I had messed up the different channel updates, this should be correct.